keras/layers/recurrent.py

```diff
@@ -363,13 +372,11 @@ def __init__(self, cell,
 
         self.supports_masking = True
         self.input_spec = [InputSpec(ndim=3)]
-        if hasattr(self.cell.state_size, '__len__'):
-            self.state_spec = [InputSpec(shape=(None, dim))
-                               for dim in self.cell.state_size]
-        else:
-            self.state_spec = InputSpec(shape=(None, self.cell.state_size))
+        self.state_spec = None
```

Comment:
For more advanced cells (typically attention cells such as this one: https://github.com/andhus/keras/pull/3/files#diff-9c4188ddc4dd173d80f64feed5b89412R258) the `cell.state_size`is not defined until the cell is built. But that's not a problem, `state_spec`is just set a bit later in build (or in `__call__` if `initial_state`is passed to `RNN`)